clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs

```diff
@@ -0,0 +1,151 @@
+    ) = gmiiSgmiiBridgePrim clockP clockN refRst signalDetect
+      (toConfigVector <$> bridgeConfig) anConfig anRestart lvdsInP lvdsInN gmiiTxD
+      gmiiTxEn gmiiTxEr
+
+  bridgeRst125 = unsafeFromActiveHigh activeHighRxRst
+  bridgeLvdsOut = Lvds <$> lvdsOutP <*> lvdsOutN
+  (DiffClock clockP clockN) = refClk
+  lvdsInP = lvdsP <$> lvdsIn
+  lvdsInN = lvdsN <$> lvdsIn
+  gmiiTxD = gmiiData <$> gmiiTx
+  gmiiTxEn = gmiiValid <$> gmiiTx
+  gmiiTxEr = gmiiError <$> gmiiTx
+  bridgeGmiiRx = Gmii <$> gmiiRxD <*> gmiiRxDv <*> gmiiRxEr
+
+-- | Black box for the GMII to SGMII bridge.
+gmiiSgmiiBridgePrim ::
+  forall async sgmii625 gmii125 .
+  ( KnownDomain sgmii625
+  , KnownDomain gmii125
+  , DomainPeriod sgmii625 ~  Picoseconds 1600
+  , DomainPeriod gmii125 ~ Nanoseconds 8
+  , DomainActiveEdge sgmii625 ~ 'Rising
+  , DomainActiveEdge gmii125 ~ 'Rising
+  , DomainResetKind async ~ 'Asynchronous
+  , DomainResetKind gmii125 ~ 'Synchronous
+  , DomainResetPolarity async ~ 'ActiveHigh
+  , DomainResetPolarity gmii125 ~ 'ActiveHigh
+  ) =>
+  -- | P channel of the reference clock coming from the PHY.
+  Clock sgmii625 ->
+  -- | N channel of the reference clock coming from the PHY.
+  ClockN sgmii625 ->
+  -- | Asynchronous active high reset for the bridge.
+  Reset async ->
+  -- | Signal detect from the PHY.
+  -- Either connect to the PHY's signal detect or use a constant @True@,
+  -- otherwise the link will never come up.
+  Signal async Bool ->
+  -- | Configuration for the bridge.
+  Signal gmii125 ConfigVector ->
+  -- | Auto negotiation configuration for the bridge.
+  Signal gmii125 AutoNegConfig ->
+
+  -- | Restart auto negotiation.
+  Signal gmii125 Bool ->
+  -- | P channel of the LVDS input from the PHY.
+  Signal sgmii625 Bit ->
+  -- | N channel of the LVDS input from the PHY.
+  Signal sgmii625 Bit ->
+  -- | GMII data input from the MAC.
+  Signal gmii125 (BitVector 8) ->
+  -- | GMII valid input from the MAC.
+  Signal gmii125 Bit ->
+  -- | GMII error input from the MAC.
+  Signal gmii125 Bit ->
+  -- |
+  -- 1. P channel of the LVDS output to the PHY.
+  -- 2. N channel of the LVDS output to the PHY.
+  -- 3. GMII data output to the MAC.
+  -- 4. GMII valid output to the MAC.
+  -- 5. GMII error output to the MAC.
+  -- 6. Clock output for the 125 MHz domain.
+  -- 7. Active high reset output for the 125 MHz domain.
+  -- 8. Status vector.
+  ( Clock gmii125
+  , Signal gmii125 Bool
+  , Signal sgmii625 Bit
+  , Signal sgmii625 Bit
+  , Signal gmii125 (BitVector 8)
+  , Signal gmii125 Bit
+  , Signal gmii125 Bit
+  , Signal gmii125 StatusVector
+  )
+
+-- Can't use @deepErrorX@ because one of the results is a clock.
+gmiiSgmiiBridgePrim !_ !_  !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ =
+  (clockGen, undefined, undefined, undefined, undefined, undefined, undefined, undefined)
+
+{-# CLASH_OPAQUE gmiiSgmiiBridgePrim #-}
+{-# ANN gmiiSgmiiBridgePrim hasBlackBox #-}
+{-# ANN gmiiSgmiiBridgePrim (verilogPrim) #-}
```

Comment:
```suggestion
-- Can't use @deepErrorX@ because one of the results is a clock.
gmiiSgmiiBridgePrim !_ !_  !_ !_ !_ !_ !_ !_ !_ !_ !_ !_ =
  (clockGen, undefined, undefined, undefined, undefined, undefined, undefined, undefined)
{-# CLASH_OPAQUE gmiiSgmiiBridgePrim #-}
{-# ANN gmiiSgmiiBridgePrim verilogPrim #-}
```


I think annotations for both `hasBlackBox` as well as `InlineYamlPrimitive` is overkill. `hasBlackBox` is more for detached primitives to make sure Clash picked them up. Let's drop `hasBlackBox`.

No blank line between definition and annotations.

You also forgot to remove the parentheses.